src/transform.js

```diff
@@ -8,37 +8,47 @@ import * as parents from "./parents.js";
  * This function will not modify the root node of the input AST.
  *
  * @param {object | object[]} node AST node or array of nodes
- * @param {Object.<string, function> | function(object, string, object?)} transformation A map of node types to callbacks, or a single callback that will be called for all node types
+ * @param {Object.<string, function> | function(object, string, object?) | (Object.<string, function> | function(object, string, object?))[]} transformations A map of node types to callbacks, or a single callback that will be called for all node types, or a list of either, which will be applied in order
  * @param {object} [o]
  * @param {string | string[] | function} [o.only] Only walk nodes of this type
  * @param {string | string[] | function} [o.except] Ignore walking nodes of these types
  * @returns {object | object[]} The transformation's return value on the root node(s) of the input AST, or the root node(s) if the transformation did not return a value
  */
-export default function transform (node, transformation, o) {
-	return _transform(node, transformation, o);
+export default function transform (node, transformations, o) {
+	if (!Array.isArray(transformations)) {
+		transformations = [transformations];
+	}
+	return _transform(node, transformations, o);
 }
 
-function _transform (node, transformation, o = {}, property, parent) {
+function _transform (node, transformations, o = {}, property, parent) {
 	if (Array.isArray(node)) {
-		return node.map(n => _transform(n, transformation, o, property, parent));
+		return node.map(n => _transform(n, transformations, o, property, parent));
 	}
 
 	const ignore = o.except && matches(node, o.except);
 	const explore = !ignore && matches(node, o.only);
 
 	if (explore) {
-		let transformedNode;
-		if (typeof transformation === "function") {
-			transformedNode = transformation(node, property, parent);
-		}
-		else if (typeof transformation === "object") {
-			transformedNode = transformation[node.type]?.(node, property, parent);
+		let transformedNode = node;
+		for (const transformation of transformations) {
+			if (typeof transformation === "function") {
+				transformedNode = transformation(transformedNode, property, parent);
+			}
+			else if (typeof transformation === "object") {
+				transformedNode = transformation[transformedNode.type]?.(transformedNode, property, parent);
```

Comment:
Just realized, since both functions have the same arguments, you could actually just get the callback, like:

```js
let callback = typeof transformation === "object" ? transformation[transformedNode.type] : transformation;
transformedNode = callback?.(transformedNode, property, parent);
```

Or even just:

```js
let callback = transformation[transformedNode.type] ?? transformation;
transformedNode = callback?.(transformedNode, property, parent);
```

Which would also allow people to provide a function to be used as a default, with properties for specific types (kind of a niche case, so I wouldn't make efforts to enable it, but if it just happens as  a side effect it's kind of nice)